### PR TITLE
Allow column resize in Environment pane

### PR DESCRIPTION
**Pull request by jmcphers**
This change makes it possible to adjust the the width of the columns in the Environment pane, useful when you're working with data that has long names (or, conversely, long values).

Closes https://github.com/rstudio/rstudio/issues/4020.

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java [hunk 1]
```diff
@@ -220,6 +226,73 @@ public void update(int index,
...
+               newWidth = 40;
+            else if (newWidth > 500)
+               newWidth = 500;
+               
+            resizer.style.width = newWidth + "px";
+         };
+
+         // Wire up event listeners
+         $wnd.addEventListener("mousemove", resize);
+         $wnd.addEventListener("mouseup", function(evt) {
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java]: Does this event still get fired if the user drag the mouse outside of the window and then performs a mouse-up? (Do we need to be cautious and de-register the handle in other cases, e.g. if focus is lost or similar?)

File: src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java [hunk 2]
```diff
@@ -220,6 +226,73 @@ public void update(int index,
...
+                      "\'" + style_.nameCol() + "'" +
+                      ");\"></div>";
+            }
+         };
+   }
+   
+   private final native void exportResizer() /*-{
+      $wnd.rstudio_beginResize = function(ele, clazz) {
+
+         // Locate the element to resize
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java]: Should we double-check that the user has clicked the primary mouse button (e.g. avoid resizing with middle-click or right-click)?

File: src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java (hunk 1 above)
Comment (jmcphers) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java]: Yes, it does!

File: src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java (hunk 2 above)
Comment (jmcphers) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java]: Yes, we should -- all taken care of now.

**Comment by catalytic-ca** [comment]
The Environment pane column widths are adjustable for the **List** view, but not for the **Grid** view.

**Pull request merged**